hw4.Rmd

```diff
@@ -22,51 +22,111 @@ The data and [documentation](http://socserv.mcmaster.ca/jfox/Books/Applied-Regre
+(@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proportion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+yes <- data %>%
+  filter(decision == "yes") %>%
+  summarise(yes <- n())
+
+prop_yes <- yes/length(data$decision)
+prop_yes
+  
 ```
 
 (@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+judge_cases <- data %>%
+  group_by(judge) %>%
+  summarise(sum_cases = n())
+
+p <- ggplot(judge_cases, aes(x = judge, y = sum_cases, fill = judge)) + geom_bar(stat = "identity") +  labs(x = "Judge", y = "Number of cases") + ggtitle("Number of cases per judge") + scale_fill_hue(c = 30, l = 80) + theme(axis.text.x = element_blank())
+
+p
+
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+
+judge_yes <- data %>%
+  group_by(judge, decision) %>%
+  filter(decision == "yes") %>%
+  summarise(yes = n())
+
+judge_cases$yes <- judge_yes$yes
+
+judge_cases <- judge_cases %>%
+  mutate(prop_yes = (yes/sum_cases))
+
+p <- ggplot(judge_cases, aes(x = judge, y = prop_yes, fill = judge)) + geom_bar(stat = "identity") + scale_fill_hue(c = 30, l = 80) + labs(x = "Judge", y = "Rate of Appeals") + ggtitle("Rate of appeals per judge") + theme(axis.text.x = element_blank())
+
+p
+  
 ```
 
-(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
+(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whether an appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+data$judge <- relevel(as.factor(data$judge), ref = "Iacobucci")
+fit <- glm(decision ~ judge, data = data, family = "binomial")
+
+table <- tidy(fit)
+table
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+#The estimate we have is in the log odds form. We need to turn it in to the probability.
+
+x <- exp(- table$estimate[1] - table$estimate[7])
```

Comment:
Try to call the estimates with reference to the judge names instead of numbers because it is more robust and less prone to error. See solutions.